static_frame/core/axis_map.py

```diff
@@ -37,31 +38,38 @@ def bus_to_hierarchy(
     Given a :obj:`Bus` and an axis, derive a :obj:`IndexHierarchy`; also return and validate the :obj:`Index` of the opposite axis.
     '''
     # NOTE: need to extract just axis labels, not the full Frame; need new Store/Bus loaders just for label data
-
     extractor = get_extractor(deepcopy_from_bus, is_array=False, memo_active=False)
 
-    tree = {}
+    def tree_extractor(index: IndexBase) -> tp.Union[IndexBase, TreeNodeT]:
+        index = extractor(index)
+        if isinstance(index, IndexHierarchy):
+            return index.to_tree()
+        return index
+
+    tree: TreeNodeT = {}
     opposite: tp.Optional[IndexBase] = None
 
     for label, f in bus.items():
         if axis == 0:
-            tree[label] = extractor(f.index)
+            tree[label] = tree_extractor(f.index)
             if opposite is None:
                 opposite = extractor(f.columns)
             else:
                 if not opposite.equals(f.columns):
+                    # What about same labels different order?
                     raise init_exception_cls('opposite axis must have equivalent indices')
         elif axis == 1:
-            tree[label] = extractor(f.columns)
+            tree[label] = tree_extractor(f.columns)
             if opposite is None:
                 opposite = extractor(f.index)
             else:
                 if not opposite.equals(f.index):
+                    # What about same labels different order?
```

Comment:
At this point we are not reindexing in any form (even just for ordering) on the opposite axis, but we might in the future.